pathfinder/PathFinder.h

```diff
@@ -30,6 +30,8 @@ class Pathfinder_client {
 public:
 	Pathfinder_client(int mf) : move_flags(mf) {  }
 	virtual ~Pathfinder_client() = default;
+	Pathfinder_client(const Pathfinder_client&) = default;
+	Pathfinder_client& operator=(const Pathfinder_client&) = default;
```

Comment:
I recommend changing these two to `= delete;` and adding two more for the move constructor/assignment:

```c++
	Pathfinder_client(const Pathfinder_client&) = delete;
	Pathfinder_client(const Pathfinder_client&) = delete;
	Pathfinder_client& operator=(Pathfinder_client&&) = delete;
	Pathfinder_client& operator=(Pathfinder_client&&) = delete;
```
The idea being that copying or moving a class with virtual members is almost always wrong and will lead to slicing.

Reply:
It does not work, because the copy constructor is used by `schedule.cc` on the derived class `Actor_pathfinder`.
```
schedule.cc:4474:27: error: call to implicitly-deleted copy constructor of 'Actor_pathfinder_client'
 4474 |                 Actor_pathfinder_client COST = (GAME_SI ? cost : cost2);
      |                                         ^      ~~~~~~~~~~~~~~~~~~~~~~~~
./paths.h:36:33: note: copy constructor of 'Actor_pathfinder_client' is implicitly deleted because base class 'Pathfinder_client' has a deleted copy constructor
   36 | class Actor_pathfinder_client : public Pathfinder_client {
      |                                 ^
./pathfinder/PathFinder.h:33:2: note: 'Pathfinder_client' has been explicitly marked deleted here
   33 |         Pathfinder_client(const Pathfinder_client&) = delete;
      |         ^
1 error generated.
```
I made the change :
```
-       Pathfinder_client(const Pathfinder_client&) = default;
-       Pathfinder_client& operator=(const Pathfinder_client&) = default;
+       Pathfinder_client(const Pathfinder_client&) = delete;
+       Pathfinder_client(Pathfinder_client&&) = delete;
+       Pathfinder_client& operator=(const Pathfinder_client&) = delete;
+       Pathfinder_client& operator=(Pathfinder_client&&) = delete;
```

Reply:
The issue above can be fixed by applying the change I suggested to `pathfinder/Pathfinder.h`, and making the following change to schedule.cc:

```diff
-		Actor_pathfinder_client COST = (GAME_SI ? cost : cost2);
+		Actor_pathfinder_client& COST = (GAME_SI ? cost : cost2);
```

I did some looking up, and there is some const-correctness changes to pathfinder stuff that allows all of these pathfinder cost classes to be passed down as const I will be doing these changes myself, so you don't have to worry.